streamer/output_stream.py

```diff
@@ -95,38 +95,24 @@ def __init__(self,
                input: Input,
                pipe_dir: str,
                codec: AudioCodec,
-               channels: int) -> None:
+               channel_layout: AudioChannelLayout) -> None:
 
     super().__init__(MediaType.AUDIO, input, codec, pipe_dir)
     # Override the codec type and specify that it's an audio codec
     self.codec: AudioCodec = codec
-
-    # TODO: Make channels an input feature instead of an output feature
-    self.channels = channels
-
-    # Until we make channels an input feature, match this output feature to a
-    # specific channel layout.  Use the first one the output channels fit into.
-    self.layout = None
-    for layout in AudioChannelLayout.sorted_values():
-      if self.channels <= layout.max_channels:
-        self.layout = layout
-        break
-
-    assert self.layout, 'Unable to find audio layout for {} channels'.format(
-        self.channels)
+    self.layout = channel_layout
 
     # The features that will be used to generate the output filename.
     self._features = {
       'language': input.language,
-      'channels': str(self.channels),
+      'channels': str(self.layout.max_channels),
```

Comment:
Would you like having this to be `'channel_layout': self.layout.get_key()` instead.
@joeyparrish